src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -36,6 +36,7 @@ public abstract class SupportObserverDataProviderBase : DiagnosticDataProvider,
             // --------------------------------------------------------------------------
             var client = new HttpClient(handler);
             client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            client.DefaultRequestHeaders.Add(HeaderConstants.UserAgentHeaderName, "appservice-diagnostics");
```

Comment:
Do you want the user-agent to somehow indicate whether the call was from external DiagnoseAndSolve or for a detector executed via AppLens? Do you think something like this will help you debug traffic/load scenarios in future?

Reply:
I think it's OK with this header, as the traffic from Applens and external DianoseAndSolve are handled by different runtimehost endpoints (euaps and non euaps) which connects to different Observer endpoints as well